src/pages/Courses/index.tsx

```diff
@@ -1,21 +1,125 @@
+                    course: this.state.inputValues.course,
+                    author: this.state.inputValues.author,
+                    stars: Number(`${this.state.inputValues.stars}`),
+                    price: Number(`${this.state.inputValues.price}`),
+                }]
+        }))
+    }
+
+    //     this.setState(({ data }) => ({ data: [...data, course] }))
+// input value={this.state.name} onChange={(value) => this.setState({ name: value })}
+
+    handleDelete = (id: number) => () => {
+        this.setState(({data}) => ({
+            data: data.filter((item) => item.id !== id)
+        }));
+    };
+
+    handleSelectCourse = (course: Course) => () => {
+        this.setState({selectedCourse: course})
+    }
+
+    handleCloseDetailsModal = () => {
+        this.setState({selectedCourse: null})
+    }
+    handleOpenModal = () => {
+        this.setState({open: true})
+    }
+    handleCloseModal = () => {
+        this.setState({open: false})
+    }
+
 
 
-class CoursePage extends React.Component<Props> {
     render() {
-        const { classes } = this.props;
+
+        const {classes} = this.props;
+        const {data, selectedCourse, open,inputValues} = this.state;
+        const course = selectedCourse as any
+
 
         return (
-            <div className={classes.courseGallery}>
-                {data.map((course) => (
-                    <CourseCard course={course} />
-                ))}
-            </div>
+            <>
+                <div className={classes.courseGallery}>
+                    {data.map((course) => (
+                        <CourseCard handleSelectCourse={this.handleSelectCourse(course)} course={course}
+                                    onDelete={this.handleDelete(course.id)}/>
+                    ))}
+                </div>
+                <Fab className={classes.fab} color="primary" aria-label="add" onClick={this.handleOpenModal}>
+                    <AddIcon/>
+                </Fab>
+                <Modal open={!!selectedCourse} onClose={this.handleCloseDetailsModal}>{!!selectedCourse && (
+                    <div className={classes.courseInfo}>
+                        <Typography variant="h6" component="h2">More Info about course</Typography>
+                        <Typography>{'Course: ' + course.course}</Typography>
+                        <Typography>{"Tagline: " + course.name}</Typography>
+                        <Typography>{"Company: " + course.author}</Typography>
+                        <Typography>{"More info about direction: " + course.info}</Typography>
+                    </div>)}</Modal>
+
+                <Modal open={open} onClose={this.handleCloseModal}>
+                    <div className={classes.inputAdd}>
+                        <Typography variant="h6" color="secondary" component="h2">Add new course</Typography>
+                        <input placeholder="Image URL" value={inputValues.img} onChange={(event) => this.setState(state=>({
```

Comment:
You can create one function for each input onChange listener.
1) Set for each input field `name` the same attribute as the name of the field in the state
2) Create function:
```
handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      const { name, value } = event.target;
      
      this.setState(state => ({ ...state, inputValues: { ...state.inputValues, [name]: value } }))
}
```
3) Use this function in `onChange`